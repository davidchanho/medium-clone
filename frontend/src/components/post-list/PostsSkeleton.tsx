import React from "react";
import PostSkeleton from "../post/PostSkeleton";

interface PostsSkeletonProps {
  amount: number;
}

function PostsSkeleton({ amount }: PostsSkeletonProps) {
  const arr = new Array(amount)
    .fill(null)
    .map((post, index) => <PostSkeleton key={`post-skeleton-${index}`} />);

  return <div className="w-100 my-2">{arr}</div>;
}

export default PostsSkeleton;
