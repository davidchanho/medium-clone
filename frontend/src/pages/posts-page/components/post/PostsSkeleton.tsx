import React from "react";
import PostSkeleton from "./PostSkeleton";

interface PostsSkeletonProps {
  amount: number;
}

function PostsSkeleton({ amount }: PostsSkeletonProps) {
  const arr = new Array(amount)
    .fill(null)
    .map((post, index) => <PostSkeleton key={`post-skeleton-${index}`} />);

  return <div>{arr}</div>;
}

export default PostsSkeleton;
