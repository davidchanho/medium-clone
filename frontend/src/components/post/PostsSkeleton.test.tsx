import React from "react";
import renderer from "react-test-renderer";
import PostsSkeleton from "./PostsSkeleton";

describe("PostsSkeleton", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<PostsSkeleton amount={3} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
