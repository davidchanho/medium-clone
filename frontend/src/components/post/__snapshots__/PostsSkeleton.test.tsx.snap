// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsSkeleton renders correctly 1`] = `
<div
  className="w-100 my-2"
>
  <div
    className="my-2 d-flex"
  >
    <span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
    </span>
    <span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={
          Object {
            "height": 135,
            "width": 200,
          }
        }
      >
        ‌
      </span>
    </span>
  </div>
  <div
    className="my-2 d-flex"
  >
    <span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
    </span>
    <span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={
          Object {
            "height": 135,
            "width": 200,
          }
        }
      >
        ‌
      </span>
    </span>
  </div>
  <div
    className="my-2 d-flex"
  >
    <span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={Object {}}
      >
        ‌
      </span>
    </span>
    <span>
      <span
        className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style={
          Object {
            "height": 135,
            "width": 200,
          }
        }
      >
        ‌
      </span>
    </span>
  </div>
</div>
`;
